src/commands.ts

```diff
@@ -87,26 +87,33 @@ async function deployCurrentProject(args, commandObserver: CommandObserver) {
 		project = args.fsPath;
 	}
 
-	await projectHelper.buildProjects([project], commandObserver).then(async result => {
-		if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
-			var filePath = await projectHelper.getOutputFilePath(project);
-			if (filePath && fs.existsSync(filePath)) {
-				var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
-				if (connection) {
-					vscode.workspace.openTextDocument(filePath).then(doc => {
-						vscode.window.showTextDocument(doc, vscode.ViewColumn.Active, false).then(() => {
-							let filePath = doc.uri.toString();
-							azdata.queryeditor.connect(filePath, connection.connectionId).then(() => azdata.queryeditor.runQuery(filePath, undefined, false));
+	try {
+		await projectHelper.buildProjects([project], commandObserver).then(async result => {
+			if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
+				await projectHelper.setOutputFilePath(project, commandObserver);
+				var filePath = commandObserver.outputFilePath;
+				if (filePath && fs.existsSync(filePath)) {
+					var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
+					if (connection) {
+						let projectFileText = fs.readFileSync(filePath, 'utf8');
+						azdata.queryeditor.connect(filePath, connection.connectionId).then(() => {
```

Comment:
So this won't open any new windows rights? Do we need to disconnect it after deploy is done? Typically they would disconnect by closing the window and without the visual - when do we close the connection?

Reply:
Yes, this would not open any new windows. Let me see if need to disconnect it explicitly